examples/platform/k32w/app/support/FreeRtosHooks.c

```diff
@@ -0,0 +1,234 @@
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+
+#include "FreeRtosHooks.h"
+
+#include "FreeRTOS.h"
+#include "semphr.h"
+
+#include <assert.h>
+
+#include <stdio.h>
+#include <string.h>
+
+#include "PWR_Interface.h"
+#include "TimersManager.h"
+#include "board.h"
+
+/* Bluetooth Low Energy */
+#include "ble_config.h"
+#include "l2ca_cb_interface.h"
+
+#include "controller_interface.h"
+
+#if defined gLoggingActive_d && (gLoggingActive_d > 0)
+#include "dbg_logging.h"
+#ifndef DBG_APP
+#define DBG_APP 0
+#endif
+#define APP_DBG_LOG(fmt, ...)                                                                                                      \
+    if (DBG_APP)                                                                                                                   \
+        do                                                                                                                         \
+        {                                                                                                                          \
+            DbgLogAdd(__FUNCTION__, fmt, VA_NUM_ARGS(__VA_ARGS__), ##__VA_ARGS__);                                                 \
+        } while (0);
+#else
+#define APP_DBG_LOG(...)
+#endif
+
+static inline void mutex_init(mbedtls_threading_mutex_t * p_mutex)
+{
+    assert(p_mutex != NULL);
+    *p_mutex = xSemaphoreCreateMutex();
+    assert(*p_mutex != NULL);
+}
+
+static inline void mutex_free(mbedtls_threading_mutex_t * p_mutex)
+{
+    assert(p_mutex != NULL);
+    assert(*p_mutex != NULL);
+    vSemaphoreDelete(*p_mutex);
+}
+
+static inline int mutex_lock(mbedtls_threading_mutex_t * p_mutex)
+{
+    assert(p_mutex != NULL);
+    assert(*p_mutex != NULL);
+    return xSemaphoreTake(*p_mutex, portMAX_DELAY) != pdTRUE;
+}
+
+static inline int mutex_unlock(mbedtls_threading_mutex_t * p_mutex)
+{
+    assert(p_mutex != NULL);
+    assert(*p_mutex != NULL);
+    return xSemaphoreGive(*p_mutex) != pdTRUE;
+}
+
+void freertos_mbedtls_mutex_init(void)
+{
+    mbedtls_threading_set_alt(mutex_init, mutex_free, mutex_lock, mutex_unlock);
+}
+
+void freertos_mbedtls_mutex_free(void)
+{
+    mbedtls_threading_free_alt();
+}
+
+#if defined(cPWR_UsePowerDownMode) && (cPWR_UsePowerDownMode) && (configUSE_TICKLESS_IDLE != 0)
+
+/*
+ * Setup the systick timer to generate the tick interrupts at the required
+ * frequency.
+ */
+void vPortSetupTimerInterrupt(void)
+{
+    /* Stop and clear the SysTick. */
+    SysTick->CTRL = 0UL;
+    SysTick->VAL  = 0UL;
+
+#if DBG_PostStepTickAssess
+    tickless_SystickCheckDriftInit();
+#endif
+
+    /* configure module for tickless mode */
+    tickless_init();
+
+#if gApp_SystemClockCalibration
+    /* start calibration - suppose 32MHz crystal is running and FRO48M running */
```

Comment:
This comment is not very clear, and I think is also possibly inaccurate.  Are we calibrating for the ~32Khz oscillator?  That's not a crystal, and not 32MHz.

Reply:
Agree, I also updated the comment. Thanks!